src/lib/probe-validator.ts

```diff
@@ -0,0 +1,49 @@
+import config from 'config';
+import TTLCache from '@isaacs/ttlcache';
+import { type RedisCluster, getMeasurementRedisClient } from './redis/measurement-client.js';
+
+export class ProbeValidator {
+	private readonly testIdToProbeId = new TTLCache<string, string>({
+		ttl: (config.get<number>('measurement.timeout') + 30) * 1000,
+	});
+
+	constructor (private readonly redis: RedisCluster) {}
+
+	addValidIds (measurementId: string, testId: string, probeUuid: string): void {
+		const key = ProbeValidator.getKey(measurementId, testId);
+		this.testIdToProbeId.set(key, probeUuid);
```

Comment:
I would instead use a regular `Map` stored at `measurementId`, with `testId` as its keys. Should, greatly reduce the amount of internal timers the TTLCache needs to deal with.